### ISSUE TITLE
fat arrow Unexpected token in the Class

### ISSUE THREAD

**Issue by iterion-mikehit** [javascript-obfuscator/javascript-obfuscator#909]
Hi, try this to obfuscatge with online obfuscator.  Any idea why error is raising? 
class alfaTest{
    constructor() {
        let test = {}
    }
    setScreen = () => {
        console.log('test');
    }
}

Raise an error

Error: ERROR at line 5: Unexpected token (5:14)
> ...setScreen = () => {...

**Comment by sanex3339** [comment]
This is because of class properties that are currently on stage 3 of ECMA proposal and does not support by acorn and escodegen

**Comment by BlackYuzia** [comment]
@iterion-mikehit you should use methods, like:

```
class Kruzya {
 // ...
 
 method () {console.log("test")}
}
```

**Comment by iterion-mikehit** [comment]
Thank you I have a more complex JS code unfortunately rewrite it

**Comment by sanex3339** [comment]
Now this proposal in stage 4. Acorn just has added the support of this feature, but `estraverse`/`escodegen` still does not support it. I'll keep this issue open.

**Comment by iterion-mikehit** [comment]
Hi thank you will test it in the moment

**Comment by sanex3339** [comment]
I forked `estraverse` too, but as the first step we have to wait until these PRs will be merged:
https://github.com/eslint/espree/pull/486
https://github.com/eslint/eslint-scope/pull/69

After that, I'll try to implement this feature in my fork of `escodegen`

**Comment by sanex3339** [comment]
PR with the support of this feature in my fork of `escodegen`
https://github.com/javascript-obfuscator/escodegen/pull/3

**Comment by sanex3339** [comment]
WIP PR to the obfuscator:
https://github.com/javascript-obfuscator/javascript-obfuscator/pull/916

Still need to wait until 
eslint/espree#486
eslint/eslint-scope#69

will be finished.

**Comment by Bdethloff** [comment]
I'm having the same problem.

I'm trying to obfuscate several classes all of which have arrow function methods.  In each file, the obfuscator reaches the first arrow function and throws this Error: ERROR at line 24: Unexpected token (24:18)

An example of the smallest class 
`import { xInit, yInit } from "../functions/displaySVG.js";
import { createSvgText} from "../functions/drawSVG.js";
import { currentUser } from "../script.js";

export default class Label {
    constructor(text, position) {
        this.text = text;
        this.coordinates = {x: xInit, y: yInit};
        this.position = position;
    }

    drawLabel = () => {
        let matrixGroup = document.getElementById("matrixGroup");
        let label = createSvgText(
            this.text, 13, "#333639",
            this.coordinates.x, this.coordinates.y,
            `mapLabel labl-${this.position} draggable`
        );
        label.setAttribute("id", `labl-${this.position}`)
        label.setAttributeNS(null, "font-weight", "bold");
        label.setAttributeNS(null, "text-decoration", "underline");

        label.addEventListener("dblclick", currentUser.maps[0].deleteLabel);

        matrixGroup.appendChild(label);
    }

    setCoordinates = (x, y) => {
        this.coordinates.x = x;
        this.coordinates.y = y;
    }
}`

**Comment by sanex3339** [comment]
PR to the @types/estree
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/52832

**Comment by Bdethloff** [comment]
Any update on this?

**Comment by sanex3339** [comment]
I linked all related PR's. We have to wait until they all will be merged.

**Comment by regnaio** [comment]
Could we please merge the PR: https://github.com/javascript-obfuscator/javascript-obfuscator/pull/916?

Thank you!

**Comment by sanex3339** [comment]
This PR cannot be merged right now. Wait until https://github.com/eslint/eslint-scope/pull/71 will be merged

**Comment by sanex3339** [comment]
Released as `2.18.0`
